spec/lib/crawler_manager_spec.rb

```diff
@@ -101,13 +118,43 @@
     end
   end
 
-  describe '#cleanup_all_processes' do
+  describe '.cleanup_all_processes' do
     let(:driver) { double }
 
-    it 'kills all phantomjs processes' do
-      allow(crawler_manager).to receive(:running_phantomjs_pids).and_return([10])
-      expect(crawler_manager).to receive(:kill_process).with(10)
-      crawler_manager.cleanup_all_processes
+    context 'There are some phantomjs processes running' do
+      let(:pids) { [10, 11] }
+      before do
+        allow_any_instance_of(Grell::CrawlerManager::PhantomJSManager).to receive(:running_phantomjs_pids).and_return(pids)
+      end
+
+      it 'logs processes pids' do
+        expect(Grell.logger).to receive(:warn).with('GRELL. Killing PhantomJS processes: [10, 11]')
+        expect(Grell.logger).to receive(:warn).with('GRELL. Sending KILL to PhantomJS process 10')
+        expect(Grell.logger).to receive(:warn).with('GRELL. Sending KILL to PhantomJS process 11')
+        described_class.cleanup_all_processes
+      end
+
+      it 'kills all phantomjs processes' do
+        expect_any_instance_of(Grell::CrawlerManager::PhantomJSManager).to receive(:kill_process).with(10)
+        expect_any_instance_of(Grell::CrawlerManager::PhantomJSManager).to receive(:kill_process).with(11)
+        described_class.cleanup_all_processes
+      end
+    end
+
+    context 'There are no phantomjs processes running' do
+      let(:pids) { [] }
+      before do
+        allow_any_instance_of(Grell::CrawlerManager::PhantomJSManager).to receive(:running_phantomjs_pids).and_return(pids)
+      end
+      it 'no warning is logged' do
+        expect(Grell.logger).not_to receive(:warn)
+        described_class.cleanup_all_processes
+      end
+      it 'No process is killed' do
```

Comment:
as usual :trollface: , I think rubocop is not going to be 🆒  with this lack of new lines between stuff here.